### PR TITLE
Enh: relocate spectral estimation code to analysis

### PR DESCRIPTION
**Pull request by m-nolan**
moved some code around to collect spectral estimation code w.r.t. #276

### CODE REVIEW

File: aopy/analysis.py [hunk 1]
```diff
@@ -1010,3 +1011,124 @@ def linear_fit_analysis2D(xdata, ydata, weights=None, fit_intercept=True):
...
+        nw (_type_, optional): _description_. Defaults to None.
+        bw (_type_, optional): _description_. Defaults to None.
+        adaptive (bool, optional): _description_. Defaults to False.
+        jackknife (bool, optional): _description_. Defaults to False.
+        sides (str, optional): _description_. Defaults to 'default'.
+
+    Returns:
+        fxx (np.array): spectrogram frequency array
+        txx (np.array): spectrogram time array
+        Sxx (np.array): multitaper spectrogram estimate
```
Comment (leoscholl) [inline aopy/analysis.py:1035]: could you include the expected shape of these outputs instead of `np.array`?

File: aopy/analysis.py [hunk 2]
```diff
@@ -1010,3 +1011,124 @@ def linear_fit_analysis2D(xdata, ydata, weights=None, fit_intercept=True):
...
+    Args:
+        data (_type_): _description_
+        fs (_type_): _description_
+        win_t (_type_): _description_
+        step_t (_type_): _description_
+        nw (_type_, optional): _description_. Defaults to None.
+        bw (_type_, optional): _description_. Defaults to None.
+        adaptive (bool, optional): _description_. Defaults to False.
+        jackknife (bool, optional): _description_. Defaults to False.
+        sides (str, optional): _description_. Defaults to 'default'.
```
Comment (leoscholl) [inline aopy/analysis.py:1030]: fill in the `_type_` and `_description_`

File: tests/test_analysis.py [hunk 3]
```diff
@@ -434,7 +442,51 @@ def test_linear_fit_analysis(self):
...
+        plt.ylabel('Band Power')
+        fname = 'lfp_bandpower.png'
+        savefig(write_dir, fname) # Should have power in [600, 312, 2000] Hz but not 10 or 4000
+
+        f, psd_filter, mu = aopy.analysis.get_psd_multitaper(self.x2, self.fs)
+        self.assertEqual(psd_filter.shape[1], self.n_ch)
+        print(mu.shape)
+        lfp = aopy.analysis.multitaper_lfp_bandpower(f, psd_filter, bands, False)
+        self.assertEqual(lfp.shape[1], self.x2.shape[1])
+        self.assertEqual(lfp.shape[0], len(bands))
```
Comment (leoscholl) [inline tests/test_analysis.py:489]: please include a test for `get_sgram_multitaper()`

**Comment by m-nolan** [comment]
@leoscholl added all the changes you requested and fixed a few bugs!

I've linked through every call this would break, but we should make an announcement about the relocation so people don't get confused.

File: aopy/analysis.py [hunk 4]
```diff
@@ -1010,3 +1011,144 @@ def linear_fit_analysis2D(xdata, ydata, weights=None, fit_intercept=True):
...
+
+    Compute multitaper estimate from multichannel signal input.
+
+    Args:
+        data (nt, nch): nd array of input neural data (multichannel)
+        fs (int): sampling rate
+        win_t (float): spectrogram window length (in seconds)
+        step_t (_type_): step size between spectrogram windows (in seconds)
+        nw (float, optional): time-half-bandwidth product. Defaults to None.
+        bw (_type_, optional): spectrogram bandwidth. Defaults to None.
```
Comment (leoscholl) [inline aopy/analysis.py:1027]: there's still a couple of `_type_` values in here, should be `float`, probably? or is `bw` a tuple?

**Review by leoscholl — APPROVED** [review]
yeah i don't know how best to announce a change like that. just a post in coding channel on slack?

**Pull request merged**